core/src/filesystem/media/rar.rs

```diff
@@ -324,3 +324,62 @@ impl FileConverter for RarProcessor {
 		Ok(zip_path)
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::filesystem::media::tests::{get_test_rar_file_data, get_test_rar_path};
+
+	use std::fs;
+
+	#[test]
+	fn test_process() {
+		// Create temporary directory and place a copy of our mock book.rar in it
+		let tempdir = tempfile::tempdir().expect("Failed to create temporary directory");
+		let temp_rar_file_path = tempdir
+			.path()
+			.join("book.rar")
+			.to_string_lossy()
+			.to_string();
+		fs::write(&temp_rar_file_path, get_test_rar_file_data())
+			.expect("Failed to write temporary book.rar");
+		let config = StumpConfig::debug();
+
+		// We can test deletion since it's a temporary file
+		let processed_file = RarProcessor::process(
+			&temp_rar_file_path,
+			FileProcessorOptions {
+				convert_rar_to_zip: true,
+				delete_conversion_source: true,
+			},
+			&config,
+		);
+		assert!(processed_file.is_ok());
+	}
+
+	#[test]
+	fn test_rar_to_zip() {
+		// Create temporary directory and place a copy of our mock book.rar in it
+		let tempdir = tempfile::tempdir().expect("Failed to create temporary directory");
+		let temp_rar_file_path = tempdir
+			.path()
+			.join("book.rar")
+			.to_string_lossy()
+			.to_string();
+		fs::write(&temp_rar_file_path, get_test_rar_file_data())
+			.expect("Failed to write temporary book.rar");
+		let config = StumpConfig::debug();
+
+		// We have a temporary file, so we may as well test deletion also
+		let zip_result = RarProcessor::to_zip(&temp_rar_file_path, true, None, &config);
+		assert!(zip_result.is_ok());
```

Comment:
Since we are testing deletion, can we also assert the expected rar was deleted?

Reply:
Added for both rar tests.